src/project_summary.rs

```diff
@@ -33,6 +31,10 @@ impl<'a> ProjectSummary<'a> {
         cache: &'a DiskCache,
         lockfile: Option<&'a Lockfile>,
     ) -> Self {
+        let max_workers = std::env::var(NUM_CPUS_ENV_VAR_NAME)
+        .ok()
+        .and_then(|x| x.parse::<usize>().ok())
+        .unwrap_or_else(num_cpus::get);
```

Comment:
How about building the sync handler and getting the max_workers from it so it's calculated only once?

Reply:
yeah - i wasn't sure if that would be unnecessary glomming and so was getting something in quickly to PoC and figured would get your eyes on it to how it should live going forward